libs/core/src/main/java/org/elasticsearch/common/Booleans.java

```diff
@@ -128,7 +128,7 @@ public static boolean parseBooleanLenient(String value, boolean defaultValue) {
         if (value == null) {
             return defaultValue;
         }
-        return !(value.equals("false") || value.equals("0") || value.equals("off") || value.equals("no"));
+        return (value.equals("false") || value.equals("0") || value.equals("off") || value.equals("no")) == false;
```

Comment:
this probably requires a little bit head scratching. Would it be easier to read if it was series of && and each being == false? Also not sure this needs to be addressed here.

Reply:
I refactored it to a `switch`.